AsyncAwait.Task1.CancellationTokens/Program.cs

```diff
@@ -17,45 +20,72 @@ internal class Program
     /// The Main method should not be changed at all.
     /// </summary>
     /// <param name="args"></param>
-    private static void Main(string[] args)
+    private static async Task Main(string[] args)
     {
         Console.WriteLine("Mentoring program L2. Async/await.V1. Task 1");
         Console.WriteLine("Calculating the sum of integers from 0 to N.");
         Console.WriteLine("Use 'q' key to exit...");
         Console.WriteLine();
 
         Console.WriteLine("Enter N: ");
+        CancellationTokenSource src = null;
 
-        var input = Console.ReadLine();
-        while (input.Trim().ToUpper() != "Q")
+        try
         {
-            if (int.TryParse(input, out var n))
+            var input = Console.ReadLine();
+
+            while (input.Trim().ToUpper() != "Q")
             {
-                CalculateSum(n);
+                if (int.TryParse(input, out var n))
+                {
+                    if (src != null)
+                    {
+                        src.Cancel();
+                        src.Dispose();
+                    }
+
+                    src = new CancellationTokenSource();
+
+                    Task task = Task.Run(() => CalculateSum(n, src.Token), src.Token);
```

Comment:
We are passing Token as a parameter in two places: `CalculateSum` and `Task.Run`.
Is it required for `Task.Run`?

Reply:
I removed the unnecessary token parameter.